src/main/java/com/google/sps/servlets/GroceryServlet.java

```diff
@@ -0,0 +1,64 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.*;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+
+/** Servlet responsible for creating new tasks. */
+@WebServlet("/grocery-list")
+public class GroceryServlet extends HttpServlet {
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        doGet(request,response);  
+        response.sendRedirect("/grocery.html");
+    }
+        public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
```

Comment:
don't do datastore puts in a doGet

In general:

- GET requests are for fetching data to return to the page (with no side effects)
- POST requests are for side effects (e.g. submitting a form, creating/updating/deleting elements)

Would it make more sense to separate the "add a grocery item" and "list all groceries" http methods? Can you also sync up with Apoorva (who added thet asks) to make sure things are consistent?

Reply:
so by making things consistent do you mean making three separate servlets like she did? or something else

Reply:
solved